Updating securing apps and authorization documentation based on the adapter deprecation/removal plan (#21177)
keycloak/keycloak: docs/documentation/securing_apps/topics/oidc/java/jboss-adapter.adoc

ifeval::[{project_community}==true]

[WARNING]
====
This adapter is now in End-Of-Life (EOL), and it is no longer supported nor receiving security updates. It will be removed in future releases.

> We recommend that you switch to Elytron OIDC Java adapter to secure your applications.
> This adapter has a similar configuration to Keycloak Java adapters and migrating your applications to it should be smooth.

Reviewer: ```suggestion
We recommend that you switch to the Elytron OIDC Java adapter to secure your applications.
This adapter has a similar configuration to Keycloak Java adapters, so you can expect a smooth migration of your applications.
```
